internal/provider/resource_ko_resolve_test.go

```diff
@@ -41,8 +38,8 @@ func TestAccResourceKoResolve(t *testing.T) {
                 }
                 `,
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestMatchResourceAttr("ko_resolve.foo", "manifests.0", regexp.MustCompile("^0: "+url+"/test-46c4b272b3716c422d5ff6dfc7547fa9@sha256:")),
-					resource.TestMatchResourceAttr("ko_resolve.foo", "manifests.1", regexp.MustCompile("^1: "+url+"/test-46c4b272b3716c422d5ff6dfc7547fa9@sha256:")),
+					resource.TestMatchResourceAttr("ko_resolve.foo", "manifests.0", regexp.MustCompile("0: "+repo.String()+"/github.com/google/ko/test@sha256:")),
```

Comment:
What's driving the namer change?  I think we want to match `ko`'s default here so that we work with DockerHub and friends, right?

Reply:
I'll back this out, since it's a needless behavior change in an already-too-big change.

But I'd like to get us to the point where we define behaviors like this at the provider level, and we can document "for Dockerhub, set provider's `namer` to `BLAH`" and have the default be full importpaths.

I don't think it's worth doing the same for `ko` since it's harder to configure setting `-P` for all invocations of `ko`, but it's easier here in IaC-land.

Reply:
`ko_build` actually defaults to `--preserve-import-paths`, which is probably not great.

Filed https://github.com/ko-build/terraform-provider-ko/issues/71 to track aligning these and making a provider-wide default.